asdf/_display.py

```diff
@@ -236,7 +235,7 @@ def _render_node(self, info, active_depths, is_tail):
         )
 
         if info.info is not None:
-            line = line + format_faint(format_italic(" # " + info.info))
+            line = line + format_faint(format_italic(" # " + info.info.strip()))
```

Comment:
This change is to deal with `info` that contains newlines like the [exposure](https://github.com/spacetelescope/rad/blob/186144b27f05bf654b5cab2331fb66d15f932cb1/src/rad/resources/schemas/exposure-1.0.0.yaml#L7) in RAD. See https://github.com/asdf-format/asdf/issues/1686#issuecomment-1814754907

Reply:
tldr; yaml is messy, I think the schemas would be better off not using literal `|` blocks and the other multiline options seems to strip embedded newlines

I think this is already being done (in some places). For example:
```
  │ │ ├─obs_id (str): 0000101001001001001011010001 # Programmatic observation identifier. The format is 'PPPPPCCAAASSSOOOVVVggsaaeeee' where 'PPPPP' is the Program, 'CC' is the execution plan, 'AAA' is the pass, 'SSS' is the segment, 'OOO' is the Observation, 'VVV' is the Visit, 'gg' is the visit file group, 's' is the visit file sequence, 'aa' is the visit file activity, and 'eeee' is the exposure ID. The observation ID is the complete concatenation of visit_id + visit_file_statement (visit_file_group + visit_file_sequence + visit_file_activity) + exposure.
```
Is pulling in the info from this multi-line title here: https://github.com/spacetelescope/rad/blob/main/src/rad/resources/schemas/observation-1.0.0.yaml#L10-L16

I think the issue is the difference between:
```yaml
this: description
        with 2 lines
```
and
```yaml
this: |
    description
    with 2 lines
```
The first is read by pyyaml as a single string with no newlines:
```python
{'this': 'description with 2 lines'}
```
whereas the second preserves the embedded and trailing newlines
```python
{'this': 'description\nwith 2 lines\n'}
```

I think that the use of these literal blocks (with the `|`) might be considered bugs in the schema (these could be 'folded', see below, or multiline formatted like the first example).

To further complicate things the yaml "folded" style retains only the trailing newline:
```yaml
this: >
    description
    with 2 lines
```
```python
{'this': 'description with 2 lines\n'}
```
Although this can be controlled using a "Block Chomping Indicator"
```yaml
this: >-
    description
    with 2 lines
```
```python
{'this': 'description with 2 lines'}
```

Reply:
I'm in favor of that change or something similar to truncate long info strings. I just checked and the "info" portion of the line is included in the "max_cols" check which truncates long lines.

I dropped the `strip` change from this PR to keep it focused on fixing the `Time` issue and opened a separate issue to track the line formatting: https://github.com/asdf-format/asdf/issues/1712